### DISCUSSION TITLE
Question regarding using Github actions on public repos.

### DISCUSSION THREAD

**Discussion by outsidephy** [community/community#30816] [Actions]
My org wants to use our CI tool on our public repos. I have a couple of questions regarding its safety.

1. Will anyone be able to trigger the workflows? This is an issue as the workflows kick off CI tests which use AWS minutes. 
2. If so, is there a way to set our workflows so they can only be run by our org members?

**Comment by Simran-B** [comment]
It depends on the event and how the workflow is set up. Manually running a workflow with `workflow_dispatch` requires write access to the repo. This is also necessary to push commits to the repo that may trigger `on: push` workflows. With `on: pull_request` it's different. You should check whether the head branch is in the same repo and not in a fork, so that the workflow doesn't run whenever someone external opens a PR in your repo.

**Reply by outsidephy** [reply]
That's helpful, how about for issue comments?
 This workflow currently uses:
 `on:
  pull_request:
    types: [opened, reopened, edited, synchronize, closed, labeled]
  issue_comment:
    branches: '**'` 

Will non-org members be able to trigger this workflow via an issue comment, if so how could we protect that?
